.github/workflows/push.yml

```diff
@@ -0,0 +1,57 @@
+name: Single Build & Test (Push)
+
+# workflow is for a branch push only and ignores master.
+# push to master (which is also pull requet merge) has a more elaborate workflow to run
+# github repo is configured with branch protection on master.
+on: 
+  push:
+    branches-ignore:
+      - 'master'
+
+  workflow_dispatch:
+    inputs: 
+      node:
+        required: false
+        description: 'A node version (e.g 10.16.0)'
+        default: '14' # TODO: update to 16 when support is added
+
+jobs:
+  build-single-test:
+    runs-on: ubuntu-latest # environment job will run in
+
+    env:
+      # TODO: move to staging when AO-20147
+      AO_TOKEN_PROD: ${{ secrets.AO_TOKEN_PROD }} 
+
+    steps:
+      - name: Checkout ${{ github.ref }}
+        uses: actions/checkout@v2
+
+      - name: Setup Node ${{ github.event.inputs.node || github.event.inputs.node.default }}
+        uses: actions/setup-node@v2
+        with:
+          node-version: ${{ github.event.inputs.node || github.event.inputs.node.default }}
+
+      - name: Show Info
+        run: |
+          whoami
+          id
+          printenv
+          node --version
+          npm --version 
+          cat /etc/os-release
+          pwd
+          ls -al
```

Comment:
I think most of the output is probably not necessary for diagnosis and might just pollute the logs. The things which I would consider removing are: `whoami`, `id`, `printenv`, `pwd` and `ls -al` as they seem to be mostly helpful for the development of the workflow itself but not so much for checking failed builds/ tests.

Reply:
The gist is that this is “Build Environment Info” (should probably change step name). Maybe ‘printenv’ stays here.

We can (or should??) also have a “Build Result Info”.

@cheempz - want to confirm/show how ‘node-pre-gyp’ got the binary (here, and/or in all future workflows ?) or are we ok with a “pass if no fail” approach?

Reply:
I'm not as concerned with checking how we got the binary in this workflow because the npm steps being run below would only allow the "build from source" scenario (well, unless we redefined what `rebuild` does).  The current steps LGTM, `npm test` would ensure that the binaries are present and working.